src/main/java/org/kohsuke/github/GHBranchProtectionBuilder.java

```diff
@@ -50,8 +50,27 @@ public class GHBranchProtectionBuilder {
      *            the checks
      * @return the gh branch protection builder
      */
+    public GHBranchProtectionBuilder addRequiredChecksWithAppIds(Collection<GHBranchProtection.Check> checks) {
+        if (!(getStatusChecks() instanceof StatusChecksWithAppId)) {
+            statusChecks = new StatusChecksWithAppId();
+        }
+        ((StatusChecksWithAppId) getStatusChecks()).checks.addAll(checks);
+        return this;
+    }
+
+    /**
+     * Add required checks gh branch protection builder.
+     *
+     * @param checks
+     *            the checks
+     * @return the gh branch protection builder
+     */
+    @Deprecated
     public GHBranchProtectionBuilder addRequiredChecks(Collection<String> checks) {
-        getStatusChecks().contexts.addAll(checks);
+        if (!(getStatusChecks() instanceof StatusChecksDeprecated)) {
+            statusChecks = new StatusChecksDeprecated();
+        }
+        ((StatusChecksDeprecated) getStatusChecks()).contexts.addAll(checks);
```

Comment:
If someone calls both `addRequiredChecks` and `addRequiredStatusChecks`, only the last call will win.  
How about something like this:
```suggestion
        getStatusChecks(StatusChecksDeprecated.class).contexts.addAll(checks);
```

Reply:
https://github.com/hub4j/github-api/pull/1844#discussion_r1633525350